### ISSUE TITLE
Did not deal with the problem of overlapping marks with same name 

### ISSUE THREAD

**Issue by dongtc** [yjs/y-prosemirror#34]
**Describe the bug**
Prosemirror supports overlapping marks with same name(as long as they have different attributes):

> excludes: ?⁠string
> Determines which other marks this mark can coexist with. Should be a space-separated strings naming other marks or groups of marks. When a mark is added to a set, all marks that it excludes are removed in the process. If the set contains any mark that excludes the new mark but is not, itself, excluded by the new mark, the mark can not be added an the set. You can use the value "_" to indicate that the mark excludes all marks in the schema.Defaults to only being exclusive with marks of the same type. **You can set it to an empty string (or any string not containing the mark's own name) to allow multiple marks of a given type to coexist (as long as they have different attributes)**.

https://prosemirror.net/docs/ref/#model.MarkSpec.excludes

But prosemirror-binding seems didn't handle it well, i have found the problem code line: 
https://github.com/yjs/y-prosemirror/blob/25cea84874eace50745fd2433847aabceef92b65/src/plugins/sync-plugin.js#L670

If a prosemirror text with the marks:`[{type: 'comment', attrs: {id: 1}}, {type: 'comment', attrs: {id: 2}}]`

It will be convert to:`{comment: {id: 2}}`

You can see lost a mark.

**Comment by dmonad** [comment]
Right.. that makes sense. But this type of information can currently not be properly represented in Yjs / Y.Text.

**Comment by dongtc** [comment]
In my project, i have a temporary solution. The main code is as follows:

![image](https://user-images.githubusercontent.com/13464443/101870330-59b72f80-3bbc-11eb-9169-64a1218e5641.png)

Meantime, i change `createTextNodesFromYText` and `equalYTextPText`

It seems hack but works fine, do you have a better suggest? @dmonad

**Comment by dmonad** [comment]
Hi @dongtc , 

the approach might work, but it will break all existing bindings. I would accept a PR that is well tested and is backwards-compatible. I think it could be possible to use arrays as "attributes" when multiple marks with the same name are present.

**Comment by dongtc** [comment]
Tanks for your suggestion, i will create a PR later. @dmonad

**Comment by bdbch** [comment]
Was this ever fixed? I think I ran into the same issue with overlapping comment marks. Works fine as long as I don't reload my editor but as soon as I reload, the overlapping marks are replaced with the outermost mark taking priority.

**Comment by dmonad** [comment]
For comments, I recommend annotating ranges using relative-positions.

For now, overlapping formatting attributes is not something that I'm looking to support in the near future.

**Comment by dmonad** [comment]
y-prosemirror@1.3.0 supports overlapping marks now (backwards compatible, support for concurrently adding overlapping marks).

**Comment by dmonad** [comment]
Oh thanks @disic and Zendis for sponsoring this work <3
